docs/security/external-principals-aws.md

```diff
@@ -143,21 +143,41 @@ To install the required packages, run the following command:
   pip install lakefs[aws-iam]
 ```
 
-In order to generate a lakeFS client with the assumed role, initiate a boto3 session with the desired role and call `lakefs.client.frow_aws_role`:
-
-
-```python
-import lakefs
-import boto3    
-session = boto3.Session()
-my_client = lakefs.client.from_aws_role(session=session, ttl_seconds=7200, host="<lakefs-host>")
-
-# list repositories
-repos = lakefs.repositories(client=my_client)
-for r in repos:
-    print(r)
-```
-
+There are two ways in which external principals can be used to authenticate to lakeFS:
+
+1. If no other authentication flow is provided, and the `credentials.provider.type` configuration is set to `aws_iam` in `.lakectl.yaml`, the client will use the machine's AWS role to authenticate with lakeFS:
+    
+   ```yaml
+    credentials:
+      provider:
+        type: aws_iam
+        aws_iam:
+          token_ttl_seconds: 3600            # TTL for the temporary token (default: 3600)
+          url_presign_ttl_seconds: 60        # TTL for presigned URLs (default: 60)
+          token_request_headers:             # Optional headers for token requests
+            HeaderName: HeaderValue
+    ```
+    Or using environment variables:
+    ```bash
+    export LAKECTL_CREDENTIALS_PROVIDER_TYPE="aws_iam"
+    export LAKECTL_CREDENTIALS_PROVIDER_AWS_IAM_TOKEN_TTL_SECONDS="3600"
+    export LAKECTL_CREDENTIALS_PROVIDER_AWS_IAM_PRESIGNED_URL_TTL_SECONDS="60"
+    export LAKECTL_CREDENTIALS_PROVIDER_AWS_IAM_TOKEN_REQUEST_HEADERS='{"HeaderName":"HeaderValue"}'
+    ```
```

Comment:
Nit: for completeness of the new method (allowing global client) please add 
"import lakefs" or mention that its global. 

Nit 2: add warn that using lakectl.yaml config is not backward compatible with older lakcectl versions.

Reply:
done